crates/spartan-farmer/src/commands/dht/client.rs

```diff
@@ -0,0 +1,308 @@
+            .await
+            .unwrap();
+
+        let result = recv
+            .await
+            .expect("Failed to retrieve the list of all known peers.");
+
+        result
+    }
+
+    // Get the list of all addresses we are listening on.
+    #[allow(dead_code)]
+    pub async fn listeners(&mut self) -> Vec<Multiaddr> {
+        let (sender, recv) = oneshot::channel();
+
+        self.client_tx
+            .send(ClientEvent::Listeners { sender })
+            .await
+            .unwrap();
+
+        let addrs = recv
+            .await
+            .expect("Failed to retrieve the list of all known peers.");
+
+        addrs
+    }
+
+    // Dial another node using Peer Id and Address.
+    #[allow(dead_code)]
+    pub async fn dial(&mut self, peer: PeerId, addr: Multiaddr) {
+        let (sender, recv) = oneshot::channel();
+
+        self.client_tx
+            .send(ClientEvent::Dial { addr, peer, sender })
+            .await
+            .unwrap();
+
+        let _ = recv.await;
+    }
+
+    // Returns the list of all the peers the client has in its Routing table.
+    #[allow(dead_code)]
+    pub async fn known_peers(&mut self) -> Vec<PeerId> {
+        let (sender, recv) = oneshot::channel();
+
+        self.client_tx
+            .send(ClientEvent::KnownPeers { sender })
+            .await
+            .unwrap();
+
+        let peers = recv
+            .await
+            .expect("Failed to retrieve the list of all known peers.");
+
+        peers
+    }
+
+    // Set listening address for a particular Normal node.
+    pub async fn start_listening(&mut self, addr: Multiaddr) {
+        // The oneshot channel helps us to pass error messages related to
+        // SwarmEvent/KademliaEvent.
+        let (sender, recv) = oneshot::channel();
+
+        self.client_tx
+            .send(ClientEvent::Listen { addr, sender })
+            .await
+            .unwrap();
+
+        // Check if the ListenEvent was processed, properly.
+        let _ = recv.await.expect("Failed to start listening.");
+    }
+
+    // Sync with other peers on the DHT. (GetClosestPeer)
+    #[allow(dead_code)]
+    pub async fn random_walk(&mut self, key: Option<PeerId>) -> QueryId {
+        let (sender, recv) = oneshot::channel();
+
+        self.client_tx
+            .send(ClientEvent::RandomWalk { key, sender })
+            .await
+            .unwrap();
+
+        // Check if the Bootstrap was processed, properly.
+        recv.await.expect("Failed to Random Walk.")
+    }
+
+    // Bootstrap
+    pub async fn bootstrap(&mut self) -> QueryId {
+        let (sender, recv) = oneshot::channel();
+
+        self.client_tx
+            .send(ClientEvent::Bootstrap { sender })
+            .await
+            .unwrap();
+
+        // Check if the Bootstrap was processed, properly.
+        recv.await.expect("Failed to bootstrap.")
+    }
+
+    pub fn handle_client_event(eventloop: &mut EventLoop, event: ClientEvent) {
```

Comment:
If method doesn't have `self` argument and doesn't produce `Self` it shouldn't really be a method in that case.

Reply:
This method is just here to seperate the logic for handling client events. Just how we have `handle_network_event` for `SwarmEvent`.

Reply:
I understand that, but it doesn't belong to this struct and will hopefully be unnecessary in the end.

Reply:
How about I move `ClientEvent` and `handle_client_event` to `eventloop.rs` ? I think, that should resolve this.

Reply:
If event loop is eliminated this shouldn't be necessary